src/rules/no-extraneous-dependencies.js

```diff
@@ -1,16 +1,22 @@
 import path from 'path'
+import fs from 'fs'
 import readPkgUp from 'read-pkg-up'
 import minimatch from 'minimatch'
 import importType from '../core/importType'
 import isStaticRequire from '../core/staticRequire'
 
-function getDependencies(context) {
+function getDependencies(context, packagePath) {
   try {
-    const pkg = readPkgUp.sync({cwd: context.getFilename(), normalize: false})
-    if (!pkg || !pkg.pkg) {
+    const pkg = packagePath
+      ? JSON.parse(fs.readFileSync(packagePath, 'utf8'))
```

Comment:
```
...
  } catch (e) {
    if (packagePath && e.code === 'ENOENT') {
      context.report({ message: 'The package.json file could not be find.' })
    }
    if (packagePath && e instanceof SyntaxError) {
      context.report({ message: 'The package.json file could not be parsed due to syntax errors.' })
    }

    return null
  }
...
```

doesn't work because `context.report(` needs a `loc` or `node`, cc: http://eslint.org/docs/developer-guide/working-with-rules#contextreport

But maybe I can throw the error in those cases?